design/clusters.go

```diff
@@ -13,6 +13,36 @@ var clusterList = JSONList(
 	nil,
 	nil)
 
+// singleCluster represents a single cluster object
+var createCluster = JSONSingle(
+	"Cluster",
+	"Holds the response to a single cluster request",
+	createClusterData,
+	nil)
+
+var createClusterData = a.Type("createClusterData", func() {
+	a.Attribute("name", d.String, "Cluster name")
+	a.Attribute("api-url", d.String, "API URL")
+	a.Attribute("console-url", d.String, "Web console URL")
+	a.Attribute("metrics-url", d.String, "Metrics URL")
+	a.Attribute("logging-url", d.String, "Logging URL")
+	a.Attribute("app-dns", d.String, "User application domain name in the cluster")
+	a.Attribute("type", d.String, "Cluster type. Such as OSD, OSO, OCP, etc")
+	a.Attribute("capacity-exhausted", d.Boolean, "Cluster is full if set to 'true'")
+
+	a.Attribute("service-account-token", d.String, "Decrypted cluster wide token")
+	a.Attribute("service-account-username", d.String, "Username of the cluster wide user")
+	a.Attribute("token-provider-id", d.String, "Token provider ID")
+	a.Attribute("auth-client-id", d.String, "OAuth client ID")
+	a.Attribute("auth-client-secret", d.String, "OAuth client secret")
+	a.Attribute("auth-client-default-scope", d.String, "OAuth client default scope")
+
+	a.Required("name", "api-url", // other URLs are optional, they can be derived from the `api-url` if not explicitely provided
+		"app-dns", "type",
+		"service-account-token", "service-account-username", "token-provider-id", "auth-client-id", "auth-client-secret",
+		"auth-client-default-scope")
+})
```

Comment:
IMPO,  we should user  `-` -> `_` e.g. `auth-client-secret` -> `auth_client_secret`, as we are already at initial stage and let's have uniform way, so everywhere we'll use `_`. If you agree on it I will update existing code.

cc @alexeykazakov

Reply:
I thought we agreed on `-` last week ? See the whole content of https://github.com/fabric8-services/fabric8-cluster/blob/79d13b3f4773af3e71bbbeb8eebf5c6b631a044d/design/clusters.go

Reply:
yes, we agreed on `-`. I know I have preference `_` over `-`. Just wanted to bring alexey's attention

Reply:
@dipak-pawar yes, consistency across the API design is what matters the most to me too (I have a personal preference for `-` over `_` but it's a personal taste, of course). 
Now, if we decide to change the other endpoints' attributes pattern from `-` to `_`, we'll also need to redeploy all the other services with an updated client lib at the same time, otherwise all requests will fail. So, for the sake of simplicity, I suggest we carry on with `-` to avoid troubles with the exists deployments. Ok for you?

Reply:
Guys, just checked and we are very bad in consistency :) We use `_`, `-`, `camelCase` all over the place. I mean in all our services. Not just here.
I do not have any preferences but let's pick something and try to stick with that. BTW is there any guidelines, good practices, etc, or it's just pure taste thing?